### PR TITLE
Add scastie link and support for list of primitives in typesafe

### PR DESCRIPTION
**Pull request by afsalthaj**

### CODE REVIEW

File: core/src/main/scala/zio/config/Config.scala [hunk 1]
```diff
@@ -53,7 +53,7 @@ object Config {
   def fromPropertyFile[A](
     filePath: String,
     configDescriptor: ConfigDescriptor[String, String, A]
-  ): ZIO[Any, Throwable, Config[A]] =
+  ): Task[Config[A]] =
```
Comment (jdegoes) [inline core/src/main/scala/zio/config/Config.scala]: Technically this should be run on the blocking thread pool.

File: magnolia/src/test/scala/zio/config/magnolia/AutomaticConfigDescriptorSpec.scala [hunk 2]
```diff
@@ -120,4 +120,10 @@ object AutomaticConfigTestUtils {
       case (None, Some(v2))     => partialMyConfig + (("anotherDefault", v2.toString))
       case (None, None)         => partialMyConfig
     }
+
+  def genAlpha: Gen[Random, String] =
+    for {
+      n <- Gen.int(1, 10)
+      s <- Gen.listOfN(n)(Gen.char(65, 122))
```
Comment (jdegoes) [inline magnolia/src/test/scala/zio/config/magnolia/AutomaticConfigDescriptorSpec.scala]: There may be alphaNumeric Gen already

Comment (afsalthaj) [reply magnolia/src/test/scala/zio/config/magnolia/AutomaticConfigDescriptorSpec.scala]: Yea, Unfortunately it emitted 03 and gave and got parsed as an `Int`.

File: typesafe/src/main/scala/zio/config/typesafe/TypeSafeConfigSource.scala [hunk 3]
```diff
@@ -114,15 +113,37 @@ object TypeSafeConfigSource {
...
+                                          Right(::(h.unwrapped().toString, t.map(_.unwrapped().toString)))
+
+                                        case _ => Left(singleton(ReadError.missingValue[Vector[String], String](path)))
+
+                                      }
+                                    }
+                                }
                         } yield res
                       } else {
                         if (parentConfig.getValue(head).valueType() == ConfigValueType.OBJECT) {
```
Comment (afsalthaj) [inline typesafe/src/main/scala/zio/config/typesafe/TypeSafeConfigSource.scala]: I will need to revisit, however, there is just about lots of `Object` operations in type safe library. So any checks here probably make it much more robust.

File: typesafe/src/main/scala/zio/config/typesafe/TypeSafeConfigSource.scala [hunk 4]
```diff
@@ -114,15 +113,37 @@ object TypeSafeConfigSource {
...
+                                        case h :: t
+                                            if (::(h, t).forall(
+                                              t =>
+                                                t.valueType() != ConfigValueType.NULL ||
+                                                  t.valueType() != ConfigValueType.LIST ||
+                                                  t.valueType() != ConfigValueType.OBJECT
+                                            )) =>
+                                          Right(::(h.unwrapped().toString, t.map(_.unwrapped().toString)))
+
+                                        case _ => Left(singleton(ReadError.missingValue[Vector[String], String](path)))
```
Comment (afsalthaj) [inline typesafe/src/main/scala/zio/config/typesafe/TypeSafeConfigSource.scala]: if user provides things like

```
users = [jon, 1, {hi = bi }]
```
it actually doesn't make sense. But the error should be fatal error providing a custom message. And not missing value. My bad.

**Comment by afsalthaj** [comment]
@jdegoes Could you approve again for this one?

**Pull request merged**